staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -309,15 +338,33 @@ const (
 // WaitForNamedCacheSync is a wrapper around WaitForCacheSync that generates log messages
 // indicating that the caller identified by name is waiting for syncs, followed by
 // either a successful or failed sync.
+//
+// TODO: logcheck:context // WaitForNamedCacheSyncWithContext should be used instead of WaitForNamedCacheSync in code which supports contextual logging.
 func WaitForNamedCacheSync(controllerName string, stopCh <-chan struct{}, cacheSyncs ...InformerSynced) bool {
-	klog.Infof("Waiting for caches to sync for %s", controllerName)
+	klog.Background().Info("Waiting for caches to sync", "controller", controllerName)
 
 	if !WaitForCacheSync(stopCh, cacheSyncs...) {
-		utilruntime.HandleError(fmt.Errorf("unable to sync caches for %s", controllerName))
+		utilruntime.HandleErrorWithContext(context.Background(), nil, "Unable to sync caches", "controller", controllerName)
 		return false
 	}
 
-	klog.Infof("Caches are synced for %s", controllerName)
+	klog.Background().Info("Caches are synced", "controller", controllerName)
+	return true
+}
+
+// WaitForNamedCacheSyncWithContext is a wrapper around WaitForCacheSyncWithContext that generates log messages
+// indicating that the caller is waiting for syncs, followed by either a successful or failed sync.
+// Contextual logging can be used to identify the caller in those log messages.
+func WaitForNamedCacheSyncWithContext(ctx context.Context, cacheSyncs ...InformerSynced) bool {
+	logger := klog.FromContext(ctx)
+	logger.Info("Waiting for caches to sync")
+
+	if !WaitForCacheSync(ctx.Done(), cacheSyncs...) {
+		utilruntime.HandleErrorWithContext(ctx, nil, "Unable to sync caches")
+		return false
+	}
+
+	logger.Info("Caches are synced")
```

Comment:
lost controller name? Is the expectation to add it to the logger? Is this called anywhere in kube?

Reply:
Not yet. The commit using this is in my https://github.com/pohly/kubernetes/tree/log-client-go-tools-cache branch, for example:

https://github.com/pohly/kubernetes/blob/56b03fc6f33f01f282864d48efc684da2f0b2bd6/pkg/controller/bootstrap/bootstrapsigner.go#L168

All controllers have their name in the logger:
https://github.com/kubernetes/kubernetes/blob/98e5a701cb76efc511cdca20ea8d32de61fbf983/cmd/kube-controller-manager/app/controllermanager.go#L769

Therefore an API that doesn't force them to repeat that string is better. Other callers can do something similar or continue to use `WaitForNamedCacheSync`.

/cc @soltysh